### PR TITLE
feat: revamping a.k.a v2

### PR DESCRIPTION
**Pull request by scott-dn**
## Why?

v2

## How?

1. Apply rust safe practice https://mmapped.blog/posts/01-effective-rust-canisters.html
2. Follow https://eips.ethereum.org/EIPS/eip-721#simple-summary
3. Deprecate dip721 suffix 
4. Full integration test
5. Tooling: auto candid, linter ...

## Tickets?

- N/A

## Contribution checklist?

- [x] The commit messages are detailed
- [x] It does not break existing features (unless required)
- [x] I have performed a self-review of my own code
- [x] Documentation has been updated to reflect the changes
- [x] Tests have been added or updated to reflect the changes
- [x] All code formatting pass
- [x] All lints pass
- [x] All tests pass

## Security checklist?

- [x] Injection has been prevented (parameterized queries, no eval or system calls)
- [x] Sensitive data has been identified and is being protected properly

## Demo?

```bash
$ cd nft-v2

# generate candid
$ make candid

# build wasm
$ make build

# deploy local replica
$ make local

# integration test
$ make test
```

### CODE REVIEW

File: nft-v2/Makefile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+candid:
+	cargo run > nft-v2.did
+	didc bind -t ts nft-v2.did  > test/factory/idl.d.ts
+	echo "// @ts-nocheck" > test/factory/idl.ts
+	didc bind -t js nft-v2.did  >> test/factory/idl.ts
+
+build: candid
+	dfx stop
+	dfx start --clean --background
```
Comment (ozwaldorf) [inline nft-v2/Makefile]: We should move these 2 lines into a separate make script, just in case someone wants to use a replica that's already running

**Review by ozwaldorf — COMMENTED** [review]
Other than that, some organization of the directory, clean everything up, but that can be done in a separate PR. Really like the typescript tests, good to show usage that way

**Review by punkbit — APPROVED** [review]
@scott-dn mind providing a more descriptive PR message for merging? And then I think we should go ahead and merge. Everything else we can include separately e.g. run the integration tests in the cloud runner.

Thanks 👍

File: nft-v2/Makefile (hunk 1 above)
Comment (scott-dn) [reply nft-v2/Makefile]: cool, it should be achieved with this 7c62de86968159c165071c888846a852004f97f6

**Review by ozwaldorf — APPROVED** [review]

**Pull request merged**